File: elixir/task_example.exs
Mix.install([
  {:plug_cowboy, "~> 2.5.2"}
])

require Logger

defmodule TaskExample do
  use Plug.Router

  plug Plug.Logger
  plug :match
  plug :dispatch

  @workload 3..5

  get "/" do
    Task.start_link(&heavy_work/0)
    send_resp(conn, :ok, "")
  end

  defp heavy_work do
    timeout = Enum.random(@workload) * 1_000
    Process.sleep(timeout)
    Logger.info("done")
  end
end

Plug.Cowboy.http(TaskExample, [])
Logger.info("server started")

File: ruby/task_example.rb
require 'sinatra'

WORKLOAD = 3..5

configure do
  set :server_settings, max_threads: 5
end

helpers do
  def heavy_work
    sleep rand(WORKLOAD)
    logger.info('done')
  end
end

get '/' do
  Thread.new { heavy_work }
end
